### PR TITLE
Add desc abot discovery and update arch image

### LINKED ISSUE #1285: Add a doc to explain what does the Discovery Pod do
## Change Request

**This repository is ONLY used to solve issues related to TiDB Operator documentation.
For other issues (related to TiDB, PD, etc), please move to [other repositories](https://github.com/pingcap/).**

Please answer the following questions before submitting your issue. Thanks!

1. Describe what you find is inappropriate or missing in the existing docs.

2. Describe your suggestion or addition.

3. Provide some reference materials (documents, websites, etc) if you could.

### PR DESCRIPTION
**Pull request by KanShiori**
### What is changed, added, or deleted? (Required)
Close #1285

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v1.3 (TiDB Operator 1.3 versions)
- [x] v1.2 (TiDB Operator 1.2 versions)
- [ ] v1.1 (TiDB Operator 1.1 versions)
- [ ] v1.0 (TiDB Operator 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: 
- Other reference link(s):

### CODE REVIEW

File: zh/architecture.md [hunk 1]
```diff
@@ -29,7 +29,8 @@ TiDB 集群的编排和调度逻辑则由下列组件负责：
 * `tidb-controller-manager` 是一组 Kubernetes 上的自定义控制器。这些控制器会不断对比 `TidbCluster` 对象中记录的期望状态与 TiDB 集群的实际状态，并调整 Kubernetes 中的资源以驱动 TiDB 集群满足期望状态，并根据其他 CR 完成相应的控制逻辑；
 * `tidb-scheduler` 是一个 Kubernetes 调度器扩展，它为 Kubernetes 调度器注入 TiDB 集群特有的调度逻辑。
 * `tidb-admission-webhook` 是一个 Kubernetes 动态准入控制器，完成 Pod、StatefulSet 等相关资源的修改、验证与运维。
-
+* `discovery` 是一个用于组件间发现的服务。每一个 TiDB 集群会对应存在一个 discovery Pod，用于该集群中组件发现其他已经创建的组件。
```
Comment (csuzhangxc) [inline zh/architecture.md:32]: 要提一下到 PD 的 proxy 吗？

Comment (KanShiori) [reply zh/architecture.md:32]: 只想一句话介绍下，感觉提了 proxy 不具体解释更迷惑。。

Comment (csuzhangxc) [reply zh/architecture.md:32]: 是个问题。。。

**Review by csuzhangxc — APPROVED** [review]

File: en/architecture.md [hunk 2]
```diff
@@ -29,6 +29,7 @@ The following components are responsible for the orchestration and scheduling lo
 * `tidb-controller-manager` is a set of custom controllers in Kubernetes. These controllers constantly compare the desired state recorded in the `TidbCluster` object with the actual state of the TiDB cluster. They adjust the resources in Kubernetes to drive the TiDB cluster to meet the desired state and complete the corresponding control logic according to other CRs;
 * `tidb-scheduler` is a Kubernetes scheduler extension that injects the TiDB specific scheduling policies to the Kubernetes scheduler;
 * `tidb-admission-webhook` is a dynamic admission controller in Kubernetes, which completes the modification, verification, operation, and maintenance of Pod, StatefulSet, and other related resources.
+* `discovery` is a server for inter-components discovery. Each TiDB cluster contains a discovery Pod which is used for componets discover other existing components in the same cluster.
```
Comment (ran-huang) [inline en/architecture.md:32]:
```suggestion
* `discovery` is a service for inter-components discovery. Each TiDB cluster contains a discovery Pod which is used for the componets to discover other existing components in the same cluster.
```

**Review by ran-huang — APPROVED** [review]

**Pull request merged**